src/pocketmine/level/Position.php

```diff
@@ -122,4 +122,10 @@ public function setComponents($x, $y, $z){
 		return $this;
 	}
 
+	public function equals(Vector3 $v){
+		if($v instanceof Position){
+			return parent::equals($v) and $v->getLevel() === $this->level;
```

Comment:
Prefer using getLevel() here, some descendents of Position such as WeakPosition don't use the level field.